core/master/src/main/java/org/drftpd/commands/usermanagement/UserManagementHandler.java

```diff
@@ -1817,12 +1817,27 @@ public CommandResponse doSITE_USER(CommandRequest request) throws ImproperUsageE
 
         Session session = request.getSession();
 
-        if (request.getUser() == null) {
-            return StandardCommandManager.genericResponse("RESPONSE_530_ACCESS_DENIED");
+        // This command can be accessed by anyone, we impose restrictions here to ensure no data is leaked we do not want to be leaked
+        boolean hasPermission = false;
+
+        // Basic check
+        if (request.getUser() != null) {
+            User reqUser = session.getUserNull(request.getUser());
+            if (reqUser.equals(myUser)) {
+                hasPermission = true;
+            } else if(reqUser.isAdmin()) {
+                hasPermission = true;
+            } else {
+                if (reqUser.isGroupAdmin()) {
+                    if (myUser.isMemberOf(reqUser.getGroup())) {
+                        hasPermission = true;
+                    }
+                }
+            }
         }
```

Comment:
Could be interesting to have a clearer list of condition.
boolean myself = reqUser.equals(myUser);
boolean adminUser = reqUser.isAdmin();
boolean askedUserInSameGroupAsMyself = ....
And then an if with or conditions to grant or not the command.

Reply:
I added comments to showcase why it is allowing. I do not really see the benefit of doing this another way.